File: 프로그래머스/2/42577. 전화번호 목록/전화번호 목록.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

bool solution(vector<string> phone_book) {
    bool answer = true;
    sort(phone_book.begin(), phone_book.end());
    
    for(int i=1;i<phone_book.size();i++) {
        string s = phone_book[i];
        string prefix = phone_book[i-1];
        if(s.substr(0, prefix.length()) == prefix) {
            return false;
        }
    }
    return answer;
}

File: 프로그래머스/2/42577. 전화번호 목록/전화번호 목록.java
import java.util.*;

class Solution {
    public boolean solution(String[] phone_book) {
        Arrays.sort(phone_book);
        String prev = phone_book[0];
        Integer len = prev.length();
        for(int i=1;i<phone_book.length;i++) {
            String cur = phone_book[i];
            if (cur.length() >= len && cur.substring(0,len).equals(prev)) return false;
            else {
                prev = phone_book[i];
                len = prev.length();
            }
        }
        return true;
    }
}